### ISSUE TITLE
gitfiletree should allow package renaming

### ISSUE THREAD

**Issue by peteruhnak** [dalehenrich/filetree#207]
It is my understanding that allowing package renames (**with preserved history**) is complex task for regular filetree.

However gitfiletree (metadata-less) takes everything from git, which does understand renames and likewise can show log for the previously named version (`git log --follow`). Is this doable?

**Comment by ThierryGoubier** [comment]
Yes and no. Yes, gitfiletree sees renamings when those are tracked by git (exemple, method rename via the method versions API in the repository -- only used by AltBrowser so far), but, once you try to retrieve a version anterior to the renaming, then recovering the method / file name isn't obvious at all. The issue is the same when renaming a package... you know it has been renamed, but you don't know how to fetch a version older than the renaming point because the name has changed.
